### PR TITLE
Add utility to translate given string slice as gNMI PathElem in a schema aware way

### PR DESCRIPTION
**Pull request by g-yusufsn**
With gNMI Path 0.4, PathElem is used to describe a path. It discriminates name and key fields. With the utility added in this change, it will be possible to translate given string slice into PathElem slice in a schema aware way.

### CODE REVIEW

File: ygot/pathtranslate/pathtranslate.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/openconfig/goyang/pkg/yang"
+
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+var (
+	seperator = "/"
```
Comment (robshakir) [inline ygot/pathtranslate/pathtranslate.go]: Sorry, just spotted this nit.

Should we make this a `const`, and also `s/seperator/separator/g` in this file.

**Comment by g-yusufsn** [comment]
Thanks for catching those!

File: ygot/pathtranslate/pathtranslate.go (hunk 1 above)
Comment (g-yusufsn) [reply ygot/pathtranslate/pathtranslate.go]: Thanks for catching this. Addressed in the next commits.

**Comment by robshakir** [comment]
Thanks very much for fixing this, and all the work here!

**Pull request merged**